### PR TITLE
add multiple data support

### PR DESCRIPTION
**Pull request by SBevz**

### CODE REVIEW

**Comment by viacheslav-karnaukh** [comment]
Please update readme and examples how to use gauge component with two arcs.

**Comment by screamy** [comment]
also we need support for independent tracks(arcs)

File: src/gauge.js [hunk 1]
```diff
@@ -16,12 +16,12 @@
   var document = global.document,
     slice = Array.prototype.slice,
     requestAnimationFrame = (global.requestAnimationFrame ||
-        global.mozRequestAnimationFrame ||
```
Comment (screamy) [inline src/gauge.js]: One more time, no formatting changes please

File: src/gauge.js [hunk 2]
```diff
@@ -131,11 +131,38 @@
...
+     */
+    function normalizeMultipleValue(value, min, limit, color) {
+      // If value is single (number) - set it ro array
+      // Value should be in format [{value: number, color: 'string'}]
+      if (!Array.isArray(value)) value = [{value, color}];
+
+      var totalValue = 0;
+      var result = [];
+
+      value.forEach(function (v) {
```
Comment (screamy) [inline src/gauge.js]: do we have any updates here? logic is not the same as it was before

File: src/gauge.js [hunk 3]
```diff
@@ -237,21 +267,12 @@
           "font-size": "100%",
           "font-family": "sans-serif",
           "font-weight": "normal",
-          "text-anchor": "middle",
-          "alignment-baseline": "middle"
```
Comment (screamy) [inline src/gauge.js]: why `"alignment-baseline"` was removed

File: src/gauge.js (hunk 2 above)
Comment (SBevz) [reply src/gauge.js]: This new  logic for multiple data

File: src/gauge.js [hunk 4]
```diff
@@ -131,11 +131,30 @@
...
+     * @param {Number} limit
+     * @param {String} color
+     * @return {Array}
+     */
+    function normalizeMultipleValue(value, min, limit, color) {
+      // If value is single (number) - set it ro array
+      // Value should be in format [{value: number, color: 'string'}]
+      if (!Array.isArray(value)) value = [{value, color}];
+
+      value.map(function(v) {
```
Comment (screamy) [inline src/gauge.js]: map use to return new array but you just modifying existing array so `forEach` suites better

File: src/gauge.js [hunk 5]
```diff
@@ -131,11 +131,30 @@
...
+        var currentValue = Number(v.value);
+        if(currentValue > limit) currentValue = limit;
+        if(currentValue < min) currentValue = min;
+        v.value = currentValue;
+        return v;
+      });
+
+      return value.sort(function(a, b) {
+        return parseFloat(b.value) - parseFloat(a.value);
```
Comment (screamy) [inline src/gauge.js]: value already a number, parseFloat is useless

File: src/gauge.js [hunk 6]
```diff
@@ -169,7 +188,7 @@
           cy = GaugeDefaults.centerY;
       return {
         end: getCartesian(cx, cy, radius, endAngle),
-      	start: getCartesian(cx, cy, radius, startAngle)
+       start: getCartesian(cx, cy, radius, startAngle)
```
Comment (screamy) [inline src/gauge.js]: any reason for that change?

File: src/gauge.js [hunk 7]
```diff
@@ -261,75 +275,196 @@
...
+       */
+      function clearGaugeSections() {
+        var elements = gaugeElement.getElementsByClassName(defaultOptions.valueDialClass);
+        if (!elements.length) return false;
+        do {
+          gaugeElement.removeChild(elements[0]);
+        } while (elements.length);
+      }
+
+      function setGaugeColor(value, duration) {
```
Comment (screamy) [inline src/gauge.js]: `setGaugeColor` and `updateColor` looks like the same methods, please leave only one

File: src/gauge.js [hunk 8]
```diff
@@ -261,75 +275,196 @@
...
+                end: v.value,
+                duration,
+                step: function (val, frame) {
+                  updateSingleGaugeSection(
+                    {value: val, color: v.color},
+                    value[i].startSectionPoint
+                  );
+                }
+              });
+            }, duration * 1000 * i);
```
Comment (screamy) [inline src/gauge.js]: why do we need to multiply duration by 1000?

File: README.md [hunk 9]
```diff
@@ -101,6 +101,8 @@ cpuGauge.setValueAnimated(90, 1);
 | ```valueDialClass``` | The CSS class of the gauge's fill (value dial) (```value```) |
 | ```valueTextClass``` | The CSS class of the gauge's text (```value-text```) |
 | ```color (new)```    | An optional function that can return a color for current value  ```function(value) {}``` |
+| ```value```          | Can be integer (```value={10}```) or collection of values (```value={[value: 10, color: "red", label: "error"]}```)  |
+| ```imposeSection ``` | This value will change partition display mode. If (```imposeSection={true}```) sections will overlap each other. If (```imposeSection={false}```) sections will be rendered one after another  |
```
Comment (screamy) [inline README.md]: please find better name for our default mode and make this mode as default

File: src/gauge.js (hunk 8 above)
Comment (SBevz) [reply src/gauge.js]: duration in secconds

File: src/gauge.js (hunk 4 above)
Comment (SBevz) [reply src/gauge.js]: Done

File: src/gauge.js (hunk 5 above)
Comment (SBevz) [reply src/gauge.js]: Done

File: src/gauge.js (hunk 7 above)
Comment (SBevz) [reply src/gauge.js]: Done

File: src/gauge.js [hunk 10]
```diff
@@ -204,10 +225,14 @@
           dialClass = opts.dialClass,
           gaugeClass = opts.gaugeClass,
           gaugeColor = opts.color,
+          gefaultDuration = opts.duration || 1,
```
Comment (screamy) [inline src/gauge.js]: gefaultDuration -> defaultDuration

File: src/gauge.js [hunk 11]
```diff
@@ -169,7 +191,7 @@
           cy = GaugeDefaults.centerY;
       return {
         end: getCartesian(cx, cy, radius, endAngle),
-      	start: getCartesian(cx, cy, radius, startAngle)
+      start: getCartesian(cx, cy, radius, startAngle)
```
Comment (screamy) [inline src/gauge.js]: align properly or revert change

**Comment by screamy** [comment]
Looks good

**Pull request merged**